lib/fluent/plugin/out_elasticsearch.rb

```diff
@@ -88,6 +89,10 @@ def configure(conf)
       @remove_keys_on_update = @remove_keys_on_update.split ','
     end
 
+    if @logstash_prefix_key && @logstash_prefix_key.is_a?(String)
+      @logstash_prefix_key = @logstash_prefix_key.split '.'
```

Comment:
`@target_index_key` can be in the form "parentkey.childkey" https://github.com/uken/fluent-plugin-elasticsearch/blob/master/lib/fluent/plugin/out_elasticsearch.rb#L291

If `@logstash_prefix_key` should also be able to handle parent/child nesting, then similar logic is needed.